velox/dwio/dwrf/common/DwrfMetadata.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/dwio/dwrf/common/Common.h"
+#include "velox/dwio/dwrf/common/wrap/dwrf-proto-wrapper.h"
+
+namespace facebook::velox::dwrf {
+
+class DwrfPostScript : public PostScript {
```

Comment:
I don't feel we actually need inheritance here. Particularly for postScript, difference in two format variants are pretty small, we can just roll all properties into the same PostScript class

Reply:
Sure, my intention was to demonstrate my proposal to solve two protobuf models problem. For later classes (like Footer), the divergence will be bigger and I think we will eventually need some methods to deal with it.